src/main/scala/io/iohk/ethereum/vm/ProgramResult.scala

```diff
@@ -1,12 +1,17 @@
 package io.iohk.ethereum.vm
 
 import akka.util.ByteString
+import io.iohk.ethereum.domain.{Address, Transaction}
 
 /**
-  * Represenation of the result of execution of a contract
-  *
-  * @param returnData bytes returned by the executed contract (set by [[RETURN]] opcode)
-  * @param storage represents changes to the storage of this account
-  * @param error defined when the program terminated abnormally
-  */
-case class ProgramResult(returnData: ByteString, storage: Storage, error: Option[ProgramError])
+ * Represenation of the result of execution of a contract
+ *
+ * @param returnData bytes returned by the executed contract (set by [[RETURN]] opcode)
+ * @param storage represents changes to the storage of this account
+ * @param error defined when the program terminated abnormally
+ */
+case class ProgramResult(returnData: ByteString,
+                         storage: Storage,
+                         transactions: Seq[Transaction] = Seq(),
+                         addressesToDelete: Seq[Address] = Seq(),
+                         error: Option[ProgramError])
```

Comment:
We may want to be able to have both an error and state of the program prior to this error.